modules/020-deckhouse/docs/internal/RELEASE.md

```diff
@@ -0,0 +1,23 @@
+# DeckhouseRelease details
+
+## Available annotations
+
+| annotation                               | description                                                                        |
+|------------------------------------------|------------------------------------------------------------------------------------|
+| release.deckhouse.io/force               | Apply specified release without any checks. Force deploy.                          |
+| release.deckhouse.io/disruption-approved | Approve release with disruptive changes. Works if update.disruptionMode is Manual. |
+| release.deckhouse.io/approved            | Approve release for deployment. Works if update.mode is Manual.                    |
+
+## Disruptive release
+
+It's a release with some potentially dangerous changes (change some default value / behavior / docker -> containerd, etc)
+To handle this release, you should add disruption check logic in a release `X-1`, for example - register requirements.DisruptionFunc in init() [example](modules/402-ingress-nginx/hooks/requirements.go)
+And then in a release `X` you should add record `"disruption:$functionName": "true"` to the [requirements.json](requirements.json) file.
+For example:
+In the release 1.35.0 set disruption check logic into ingressNginx
+In the release 1.36.0 set `"disruption:ingressNginx": "true"` to the requirements.json
+
+## Release requirements
+
+For checking some precondition/requirement - register CheckFunc like [here](modules/402-ingress-nginx/hooks/requirements.go)
+and then add `"$functionName": "$version"` to the [requirements.json](requirements.json) file. (Example: `"ingressNginx": "0.33"`)
```

Comment:
```suggestion
and then add `"$functionName": "$version"` to the [requirements.json](requirements.json) file. (Example: `"disruption:ingressNginx": "0.33"`)
```

Reply:
Well, we are putting all this values into openapi values. They will look and process much more complicated, that why I decided to highlight it with the prefix

Reply:
Okay. Nesting is inconvenient. 

What I still miss is the asymmetry in notations. I got it why I am wrong here. It would be more clear to me, if we had more explicit notations. E.g.:

```json
{
  "check:ingressNginx": "1.1",
  "disruption:ingressNginx": "1.1"
}
```

What do you think?

Reply:
Let's stick to the domain and terminology a little bit. 

I guess, what we fill in `requirements.json` is not a disruption in any sense. It is a precondition check and an update requirement. We describe only the condition data and the fact of the requirement should be met:

1. precondition **check** is "1.1" for key "ingressNginx"
2. update **requirement** must be met (being "true") for "ingressNginx" to update Deckhouse

And there can be no disruption! 

- If the cluster has already had no default controller version, the check condition has already passed, and the requirement is already met. So there will be no disruption. 
- If a client notices the disruptive alert/notification and satisfies the **requirement**, there will be no disruption. 

So, I would pick more concise wording for this file.

```go
type ... interface {
    SetCoditionCheckFunc(string, func)
    Check(string)

    SetRequirementFunc(string, func)
    IsRequired(string)
}
```

```json
{
    "check:ingressNginx": "1.1",
    "require:ingressNginx": "true" // BTW why not boolean here?
}
```

What do you think?